miio/vacuum.py

```diff
@@ -596,10 +597,12 @@ def callback(ctx, *args, id_file, **kwargs):
         @dg.resultcallback()
         @dg.device_pass
         def cleanup(vac: Vacuum, *args, **kwargs):
-            if vac.ip is None:  # dummy Device for discovery, skip teardown
+            if (
+                vac.command_sender.ip is None
```

Comment:
IMO ip is a property of the device, not of the protocol.

Reply:
✅ Brought `ip` and `token` back. Token because of the `dev.token = m.checksum` in `discovery.py` (to be on the safe side).

The discovery will indeed need more work at some point.